src/visual.ts

```diff
@@ -218,43 +217,41 @@ export class AsterPlot implements IVisual {
             left: 10
         });
 
-        const rootElement = select(options.element);
+        const rootElement: d3Selection<HTMLElement, null, HTMLElement, null> = select(options.element);
 
-        const svg: Selection<any> = this.svg = rootElement
+        const svg = this.svg = rootElement
             .append("svg")
             .classed(AsterPlotVisualClassName, true)
             .style("position", "absolute");
 
         this.colorPalette = options.host.colorPalette;
         this.colorHelper = new ColorHelper(this.colorPalette);
-        this.mainGroupElement = svg.append("g");
-        this.mainLabelsElement = svg.append("g");
+        this.mainGroupElement = svg.append("g").attr("id", "mainGroup");
+        this.mainLabelsElement = svg.append("g").attr("id", "mainLabels");
 
-        this.behavior = new AsterPlotWebBehavior();
-        this.clearCatcher = appendClearCatcher(this.mainGroupElement);
+        this.behavior = new Behavior(this.colorHelper, this.selectionManager);
+        this.clearCatcher = this.mainGroupElement
+            .append("rect")
+            .classed("clearCatcher", true)
+            .attr("width", "100%")
+            .attr("height", "100%");
 
         this.slicesElement = this.mainGroupElement
             .append("g")
+            .classed(AsterPlot.AsterSlices.className, true)
             .attr("role", "listbox")
-            .attr("aria-multiselectable", "true")
-            .classed(AsterPlot.AsterSlices.className, true);
-
-        this.interactivityService = createInteractivitySelectionService(options.host);
+            .attr("aria-multiselectable", "true");
 
-        this.legend = createLegend(
-            options.element,
-            options.host && false,
-            this.interactivityService,
-            true);
+        const isScrollable: boolean = false;
+        this.legend = createLegend(options.element, isScrollable);
```

Comment:
Is it possible for legend to be scrollable? I see constant used to create it. Maybe we can remove the second param at all in this case?

Reply:
Yes, a legend is scrollable. When the area is too small for a legend, the navigation arrows are rendered, which kinda do the work of the scroll. If you pass `false` as value to `isScrollable` then the legend items will be hidden if they do not fit in the visual.

Personally, I prefer to enable the scrolling. Please, comment if you disagree.